### ISSUE TITLE
Review man pages

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1888]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)| @jml reported|
|-|-|
|Trac ID|trac#1888|
|Type|task|
|Created|2006-07-01 13:35:45Z|
|Branch|https://github.com/twisted/twisted/tree/review-man-pages-1888|

Does _anyone_ actually update manpages?

Attachments:

* [man-1888.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/41d/41d04643dd4c86a9b8f41a64a0c15b50afc83b08/5e3b30904be1f0635a87ea6c162efb824d0ff4fb.patch) (9151 bytes) - added by collab on 2008-07-05 14:50:11Z - patch against r24237

<details><summary>Searchable metadata</summary>

```
trac-id__1888 1888
type__task task
reporter__jml jml
priority__normal normal
milestone__ 
branch__branches_review_man_pages_1888 branches/review-man-pages-1888
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1151760945000000 1151760945000000
changetime__1218391584000000 1218391584000000
version__None None
owner__ 
cc__teratorn cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="collab's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|collab set owner to collab<br>collab set status to `assigned`|
|-|-|

I'll start with the most used tools (twistd, trial etc).

**Comment by twisted-trac** [comment]
|<img alt="collab's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|collab removed owner<br>collab set status to `new`|
|-|-|

Reviewed the man pages, see attached [[man-1888.patch](None/man-1888.patch) patch].

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

  * No one really uses the `--extra` parameter to `trial` anymore.  The option should probably be deprecated, then removed.  Documenting it is okay, but doesn't really add much.
  * The `--euid` option to `twistd` also sets the egid instead of the gid.  Also, it _does_ shed privileges, it just does so with `seteuid` and `setegid` instead of `setuid` and `setgid`.  I think it's a bit misleading to document it as it is documented in this patch.
  * In the `conch` man page, _received_ is misspelled as _receieved_.

Please check the patch into a branch and force a build of it on the documentation builder to make sure everything parses/etc.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra set status to `assigned`|
|-|-|

Replying to [exarkun](#note_6):
>   No one really uses the `--extra` parameter to `trial` anymore.  The option should probably be deprecated, then removed.  Documenting it is okay, but doesn't really add much.

ok, left it in the branch.

>   The `--euid` option to `twistd` also sets the egid instead of the gid.  Also, it _does_ shed privileges, it just does so with `seteuid` and `setegid` instead of `setuid` and `setgid`.  I think it's a bit misleading to document it as it is documented in this patch.

Shall I leave out the --euid option?

>   In the `conch` man page, _received_ is misspelled as _receieved_.
>
> Please check the patch into a branch and force a build of it on the documentation builder to make sure everything parses/etc.

Created a new [branches/review-man-pages-1888 branch](branches/review-man-pages-1888 branch).

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra commented|
|-|-|

(In [24429]) Branching to 'review-man-pages-1888'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_9"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

Replying to [exarkun](#note_6):
> Please check the patch into a branch and force a build of it on the documentation builder to make sure everything parses/etc.
>

Created a new branch and the doc builder reports now problems: http://buildbot.twistedmatrix.com/builders/documentation/builds/711

Putting it up for review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

I'd include `--euid`, but document it more precisely than it was in the patch.

The rest of the changes look just fine.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_11"></a>|@thijstriemstra commented|
|-|-|

Replying to [exarkun](#note_10):
> I'd include `--euid`, but document it more precisely than it was in the patch.

I tried to put your description of euid in there but I can't come up with anything that makes sense, cause I'm not familiar with the technical details behind it. If you can give me a good description I'd be happy to merge it in there.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_12"></a>|@thijstriemstra set status to `assigned`|
|-|-|

fyi: I included the original description from the patch in the branch (r24469).

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @exarkun<br>@thijstriemstra set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_14"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

On second thought, the man page now matches the output of `twistd --help`.  Maybe euid documentation should be improved too, but this ticket is just about updating the man pages.

So, looks good to me, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_15"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [24568]) Merge review-man-pages-1888: Reviewed man pages for commandline tools

Author: thijs
Reviewer: exarkun
Fixes: [#1888](https://github.com/twisted/twisted/issues/1888)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
